### DISCUSSION TITLE
Cannot change sounds

### DISCUSSION THREAD

**Discussion by jtomasrl** [probonopd/MiniDexed#472] [General]
I just picked up and old RPI3 and a few hardware I had lying around, added the files to an SD card (august release) and plugged into a launchkey mini MIDI controller through USB. It works, it sounds and I can even play chords, but when trying to modify parameters, no matter what I change, I keep getting the same sounds

Im using the sound directly from the RPI3 jack output, havent loaded any additional voice.

**Comment by diyelectromusic** [comment]
It may be that you have having the issue discussed here: https://github.com/probonopd/MiniDexed/issues/384

Basically the default configuration plays four tone generators all at once (they all listen on the OMNI MIDI channel - i.e. all of them), so if you are changing voices on one, three others keep playing the same Brass 1 voice.

As a group we collectively appear to have differing opinions on whether this is a good thing or not :)

Kevin

**Reply by donluca** [reply]
The default performance has also reverb (where available) enabled IIRC.

IMHO the default performance MiniDexed ships with should be as basic and simple as possible: only one TG enabled with BRASS1 patch (or FM Piano or whatever fallback MiniDexed uses for when there are no sysex files) and all effects disabled.

I remember being really disoriented when I first booted MiniDexed and then had to look into the Discussions to see what was happening and finally realizing that a standard performance was being loaded as default.

On the other hand, one could make a point that the first thing you play on a synth should be the best sounding one as it's always the case with various synths: you buy it, plug it in, power on and play some notes and get amazed at the default sound you're getting.
But since no one here is selling you a product... would this point still be valid?

**Reply by diyelectromusic** [reply]
> On the other hand, one could make a point that the first thing you play on a synth should be the best sounding one as it's always the case with various synths: you buy it, plug it in, power on and play some notes and get amazed at the default sound you're getting. But since no one here is selling you a product... would this point still be valid?

That's never been true on any synth I've had :)  No, the first thing I've always done is played through any preset voices to get a feel for what they can do.

And anyway, these days, who's likely to buy a synth without watching a video of someone demoing it for real?

For MiniDexed, if someone has got as far as digging out a Pi, downloading it, messing around with SD cards, possibly hooking up some hardware to drive it, then they've probably already seen Floyd Steinberg's video that is linked from the main GitHub page...

Kevin

**Comment by Banana71** [comment]
I also found my way to the miniDexed via Floyd Steinberg. Unfortunately the video is no longer up to date. - The menu navigation of the miniDexed has changed and - the performance menu has been added. If you turn on the device for the first time, with the expectations from the video, everyone would probably be totally confused. I would suspect the unexpected behavior as my own mistake during construction. E.g. with a double click you no longer get one level deeper into the menu :-(

It's best if Floyd shoots a new video again :-)
In any case, we have to point out how the miniDexed behaves when it is switched on for the first time. (Quick Start Documentation).
As a default voice I would take the one created by Floyd.


How do other synthesizers behave after a system start?
- Yamaha SY99: returns exactly to the point at which the device was switched off.
- Yamaha Motif ES: The behavior can be set in the Utility menu. (Performance, Voice, GM, last, Master)
- Yamaha Montage, MODX: The behavior can be set in the Utility menu ("Live Set" or Performance)

The miniDexed is actually always in performance mode and starts with the TG1 menu. The only thing you can set so far is the default performance.

I would find it fantastic to make the startup behavior configurable in the minidex.ini.

minidexed.ini
PowerOnMode: 0, default start to TG1 (as before)
PowerOnMode: 1, performance start to performance load
PowerOnMode: 2, E.Piano Mode, you can only choose between the Performances and do not make any changes.
PowerOnMode: 3, band mode, networking with other miniDexed via the Internet up to 5 participants are possible. (planned for the first quarter of 2024)

In Mode 2, operating errors could be almost completely ruled out. (Children, Grandma, ...)
Mode 3 is of course just for fun

**Reply by diyelectromusic** [reply]
We're not talking about general startup.  Once people know what they are doing it isn't so bad.

We're talking about the fresh-install, "out of the box", first user experience, and having a first user experience of "I've change the voice but nothing happens" is IMO very un-user friendly.

The reference to the video was not as a tutorial (and yes it ought to be noted that the interface has changed), but in response to the argument that a "fuller performance" is required on first boot to make it "sound good straight away" to get people hooked in.  I was making the point that there are videos that already show how it can sound so if the price of "sounding good on first boot" is then subsequent user confusion, then I think the negative impression of it not changing voices far outweighs any "wow it sounds good when I turn it on" feeling.

Once the basics have been observed then the menus can be explored and new performances that make it sound better can be explored.

Kevin

**Reply by Banana71** [reply]
sorry, i mentioned several things at once again

But I am convinced that Floyd's video will be used by most as a tutorial.
About the default performance:
Anyone who is familiar with the miniDexed can adjust the default performance according to their ideas or save an existing performance as default performances.

You're right, a simple Voice as default would be the least confusing.
Then please simply replace the Performance.ini, a single voice on TG1 in the MidiChannel: Omni.

An example: 
[default performance.zip](https://github.com/probonopd/MiniDexed/files/11177089/default.performance.zip)


> How do other synthesizers behave after a system start? ........

The whole second part is a different point and has nothing to do with the actual topic, but it is important for me to talk about it.

**Reply by diyelectromusic** [reply]
Yes I think you're right on both those extra things.  Maybe start a new discussion and see what people think?

I'd suggest asking someone to make a new "tutorial" video, but we keep going and changing things ;)

Kevin

**Reply by donluca** [reply]
For what is worth... after I realized what was happening when I first started using MiniDexed, I just deleted performance.ini and that was it for me.
Without the file, MiniDexed defaults to just 1 TG active with the first voice of the first bank.

We might want to distribute MiniDexed without a performance.ini (or maybe a completely empty one?)

**Reply by Banana71** [reply]
@donluca 
> We might want to distribute MiniDexed without a performance.ini (or maybe a completely empty one?)

In the default, all parameters are explained, which helped me at the beginning. I proceeded as you did and deleted the performance.ini and after the system start saved the performance as default and then reinserted and supplemented the description. The resulting performance.ini has the properties as if it weren't there at all, but has all the information that a beginner could use to help.

If there are no objections  @probonopd  please copy this performance to main. 

[performance.ini](https://github.com/probonopd/MiniDexed/files/11182914/performance.zip)


>
> performance.ini
>


> This is the miniDexed default performance and is loaded first when the system starts. TG1 receives notes from all midi channels (MIDIChannel1= >16: omni mode) and plays the first sound from the first sound bank. If there is no sound bank on the SD card, the first Voice from the DX7 Mark 1 factory bank will be loaded. With all other tone generators TG2 - TG8 the Midi channel is set to off (MIDIChannel= 0: off) and no sound is played. This performance can be replaced by your own performance at any time. (see on Wiki - Menu - Performance - Save - set as default)

> TG#
BankNumber#=0		# 0 .. 127
#VoiceNumber#=1		# 1 .. 32
#MIDIChannel#=1		# 1 .. 16, 0: off, >16: omni mode
#Volume#=100		# 0 .. 127
#Pan#=64		# 0 .. 127
#Detune#=0		# -99 .. 99
#Cutoff#=99		# 0 .. 99
#Resonance#=0		# 0 .. 99
#NoteLimitLow#=0	# 0 .. 127, C-2 .. G8
#NoteLimitHigh#=127	# 0 .. 127, C-2 .. G8
#NoteShift#=0		# -24 .. 24
#ReverbSend#=0		# 0 .. 99
#PitchBendRange#=2 # 0 .. 12
#PitchBendStep#=0 # 0 .. 12
#PortamentoMode#=0 # 0 .. 1
#PortamentoGlissando#=0 # 0 .. 1
#PortamentoTime#=0 # 0 .. 99
#VoiceData#= # space separated hex numbers of 156 voice parameters. Example: 5F 1D 14 32 63 [....] 20 55 
#MonoMode#=0 # 0-off .. 1-On
#ModulationWheelRange#=99 # 0..99
#ModulationWheelTarget#=1 # 0..7
#FootControlRange#=99 # 0..99
#FootControlTarget#=0 # 0..7
#BreathControlRange#=99 # 0..99
#BreathControlTarget#=0 # 0..7
#AftertouchRange#=99 # 0..99
#AftertouchTarget#=0 # 0..7

>TG1
BankNumber1=0
VoiceNumber1=1
MIDIChannel1=255
Volume1=100
Pan1=64
Detune1=0
Cutoff1=99
Resonance1=0
NoteLimitLow1=0
NoteLimitHigh1=127
NoteShift1=0
ReverbSend1=0
PitchBendRange1=2
PitchBendStep1=0
PortamentoMode1=0
PortamentoGlissando1=0
PortamentoTime1=0
VoiceData1=
MonoMode1=0
ModulationWheelRange1=99
ModulationWheelTarget1=7
FootControlRange1=99
FootControlTarget1=0
BreathControlRange1=99
BreathControlTarget1=0
AftertouchRange1=99
AftertouchTarget1=0

>TG2
........

**Reply by diyelectromusic** [reply]
What is the default for the MIDI channel?  OMNI is the most friendly, but only if configured on a single TG. If all are OMNI we have the same problem back again :)  So if we want an OMNI channel, then probably need to disable MIDI for all other TGs.

_Update: looking at the code, the default is MIDI disabled for all TGs, then if a performance.ini can't be loaded TG 1 is set to OMNI, so that should be fine._

A beginner friendly alternative might be default to channels 1-8 for TGs 1-8 then all 8 are available but won't get muddled into each other when played...

Kevin

**Reply by Banana71** [reply]
Ja Moin,
Actually, I wrote all of that in the post above you.

This is the miniDexed default performance and is loaded first when the system starts. TG1 receives notes from all midi channels (MIDIChannel1= >16: omni mode) and plays the first sound from the first sound bank. If there is no sound bank on the SD card, the first Voice from the DX7 Mark 1 factory bank will be loaded. With all other tone generators TG2 - TG8 the Midi channel is set to off (MIDIChannel= 0: off) and no sound is played. This performance can be replaced by your own performance at any time. (see on Wiki - Menu - Performance - Save - set as default)

**Reply by diyelectromusic** [reply]
My mistake!  I'd mistook that for comments in the performance.ini file and didn't read it properly :)

**Reply by Banana71** [reply]
@diyelectromusic ,
I had created a whole set of INIT Performances, and there is also a MultiTimbral Performance. https://github.com/probonopd/MiniDexed/discussions/325#discussioncomment-5498474
In a MultiTimbral Performance the VoiceData1-8 for TG1-TG8 should be available. Otherwise the result is not predictable if a bank is available.
For a single voice performance, the first voice from the first bank will be loaded. In an 8TG performance with different voices from the first bank, it quickly sounds caotic. Do you understand what I mean?

We should now simply replace the 4 layer Brass Performance and see how it is received and understood.
I am now at the point where, like @donluca , I would recommend the simplest possible variant (single voice performance)

Single Voice Performance with discription: [performance.ini](https://github.com/probonopd/MiniDexed/files/11182914/performance.zip)

MultiTimbral: [performance.ini multitimbral default.zip](https://github.com/probonopd/MiniDexed/files/11183189/performance.ini.multitimbral.default.zip) , please rename to performance.ini
TG1 to TG8 are active, you can Select Voice via Menue - TG1 to TG8 - Voice. MIDIChannel 1 - 8 ; all PAN are in Center

**Comment by probonopd** [comment]
@Banana71 would it be OK for you to put your INIT performance in the first slot directly after the Default performance, instead of putting it at the end? This way it could be loaded more easily via the menu.
